packages/yew/src/dom_bundle/mod.rs

```diff
@@ -0,0 +1,146 @@
+//! Realizing a virtual dom on the actual DOM
+//!
+//! A bundle, borrowed from the mathematical meaning, is any structure over some base space.
+//! In our case, the base space is the virtual dom we're trying to render.
+//! In order to efficiently implement updates, and diffing, additional information has to be
+//! kept around. This information is carried in the bundle.
+
+mod app_handle;
+mod bcomp;
+mod blist;
+mod bnode;
+mod bportal;
+mod bsuspense;
+mod btag;
+mod btext;
+
+#[cfg(test)]
+mod tests;
+
+use self::bcomp::{BComp, Scoped};
+use self::blist::BList;
+use self::bnode::BNode;
+use self::bportal::BPortal;
+use self::bsuspense::BSuspense;
+use self::btag::BTag;
+use self::btext::BText;
+
+pub(crate) use self::bcomp::{Mountable, PropsWrapper};
+pub(crate) use self::btag::{InputFields, Value};
+
+#[doc(hidden)] // Publically exported from crate::app_handle
+pub use self::app_handle::AppHandle;
+#[doc(hidden)] // Publically exported from crate::html
+pub use self::bcomp::{AnyScope, Scope, SendAsMessage};
+#[doc(hidden)] // Publically exported from crate::events
+pub use self::btag::set_event_bubbling;
+#[cfg(test)]
+#[doc(hidden)] // Publically exported from crate::tests
+pub use self::tests::layout_tests;
+
+use crate::NodeRef;
+use web_sys::{Element, Node};
+
+trait DomBundle {
+    /// Remove self from parent.
+    fn detach(self, parent: &Element);
+
+    /// Move elements from one parent to another parent.
+    /// This is for example used by `VSuspense` to preserve component state without detaching
+    /// (which destroys component state).
+    fn shift(&self, next_parent: &Element, next_sibling: NodeRef);
+}
+
+// TODO(#938): What about implementing `VDiff` for `Element`?
+// It would make it possible to include ANY element into the tree.
+// `Ace` editor embedding for example?
+
```

Comment:
Is this still valid? I was able to insert and use custom elements just fine with `material-yew`.

This is unrelated to this PR. I just saw this and wanted to bring it up

Reply:
Hm, I think not. Should be removed as explained in the linked issue.